assets/src/edit-story/components/richText/editor.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Editor } from 'draft-js';
+import PropTypes from 'prop-types';
+import {
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from 'react';
+
+/**
+ * Internal dependencies
+ */
+import useUnmount from '../../utils/useUnmount';
+import useRichText from './useRichText';
+import customInlineDisplay from './customInlineDisplay';
+
+function RichTextEditor({ content, onChange }, ref) {
+  const editorRef = useRef(null);
+  const {
+    state: { editorState },
+    actions: {
+      setStateFromContent,
+      updateEditorState,
+      getHandleKeyCommand,
+      getContentFromState,
+      clearState,
+    },
+  } = useRichText();
+
+  // Load state from parent when content changes
+  useEffect(() => {
+    setStateFromContent(content);
+  }, [setStateFromContent, content]);
+
+  // Push updates to parent when state changes
+  useEffect(() => {
+    const newContent = getContentFromState(editorState);
+    if (newContent) {
+      onChange(newContent);
+    }
+  }, [onChange, getContentFromState, editorState]);
+
+  // Set focus when initially rendered.
+  useLayoutEffect(() => {
+    if (editorRef.current) {
+      editorRef.current.focus();
+    }
+  }, []);
+
+  const hasEditorState = Boolean(editorState);
+
+  // On unmount, clear state in provider
+  useUnmount(clearState);
+
+  // Allow parent to focus editor and access main node
+  useImperativeHandle(ref, () => ({
+    focus: () => editorRef.current?.focus?.(),
+    getNode: () => editorRef.current?.editorContainer,
```

Comment:
I didn't really know what the expectation was for this kind of API - I couldn't find many examples of `useImperativeHandle` in the wild, so there doesn't seem to be a best-practice here. I can go either way.

Reply:
Just a FYI. Either way is fine by me as well. Generally `useImperativeHandle` either returns an actual element, or exposes APIs that are kind of similar to DOM elements in nature. E.g. a FancyVideo component can return the actual video element so that a caller can call `play()` and `pause()`. But YoutubeVideo component doesn't have a video element (since it's an iframe) and would often simulate video element's API.

Reply:
I kept this as-is.